sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/SchedulerRunner.java

```diff
@@ -0,0 +1,152 @@
+                        schedulerBuilder.getStateStore());
+            }
+        } finally {
+            locker.unlock();
+        }
+    }
+
+    private static void runScheduler(
+            Scheduler mesosScheduler, ServiceSpec serviceSpec, SchedulerConfig schedulerConfig, StateStore stateStore) {
+        // Install the certs from "$MESOS_SANDBOX/.ssl" (if present) inside the JRE being used to run the scheduler.
+        DcosCertInstaller.installCertificate(schedulerConfig.getJavaHome());
+
+        Protos.FrameworkInfo frameworkInfo = getFrameworkInfo(serviceSpec, stateStore);
+        LOGGER.info("Registering framework: {}", TextFormat.shortDebugString(frameworkInfo));
+        String zkUri = String.format("zk://%s/mesos", serviceSpec.getZookeeperConnection());
+        Protos.Status status = new SchedulerDriverFactory()
+                .create(mesosScheduler, frameworkInfo, zkUri, schedulerConfig)
+                .run();
+        LOGGER.error("Scheduler driver exited with status: {}", status);
+        // DRIVER_STOPPED will occur when we call stop(boolean) during uninstall.
+        // When this happens, we want to continue running so that we can advertise that the uninstall plan is complete.
+        if (status != Protos.Status.DRIVER_STOPPED) {
+            SchedulerUtils.hardExit(SchedulerErrorCode.DRIVER_EXITED);
+        }
+    }
+
+    private static Protos.FrameworkInfo getFrameworkInfo(ServiceSpec serviceSpec, StateStore stateStore) {
+        Protos.FrameworkInfo.Builder fwkInfoBuilder = Protos.FrameworkInfo.newBuilder()
+                .setName(serviceSpec.getName())
+                .setPrincipal(serviceSpec.getPrincipal())
+                .setFailoverTimeout(TWO_WEEK_SEC)
+                .setUser(serviceSpec.getUser())
+                .setCheckpoint(true);
+
+        setRoles(fwkInfoBuilder, serviceSpec);
+
+        // The framework ID is not available when we're being started for the first time.
+        Optional<Protos.FrameworkID> optionalFrameworkId = stateStore.fetchFrameworkId();
+        optionalFrameworkId.ifPresent(fwkInfoBuilder::setId);
+
+        if (!StringUtils.isEmpty(serviceSpec.getWebUrl())) {
+            fwkInfoBuilder.setWebuiUrl(serviceSpec.getWebUrl());
+        }
+
+        if (Capabilities.getInstance().supportsGpuResource()
+                && CapabilityValidator.serviceSpecRequestsGpuResources(serviceSpec)) {
+            fwkInfoBuilder.addCapabilities(Protos.FrameworkInfo.Capability.newBuilder()
+                    .setType(Protos.FrameworkInfo.Capability.Type.GPU_RESOURCES));
+        }
+
+        if (Capabilities.getInstance().supportsPreReservedResources()) {
+            fwkInfoBuilder.addCapabilities(Protos.FrameworkInfo.Capability.newBuilder()
+                    .setType(Protos.FrameworkInfo.Capability.Type.RESERVATION_REFINEMENT));
+        }
+
+        return fwkInfoBuilder.build();
+    }
+
+    @SuppressWarnings("deprecation") // mute warning for FrameworkInfo.setRole()
```

Comment:
Should we not rather handle the deprecation?

Reply:
(or at least create a ticket to address this)

Reply:
The deprecation was assigned in the mesos protobuf definition*, which defines the `FrameworkInfo` class. We still need to use the field for older mesos versions/clusters despite it being deprecated in the current version. From our perspective this warning can only be avoided by dropping support for older mesos versions.

* https://github.com/apache/mesos/blob/master/include/mesos/mesos.proto#L265

Reply:
Thanks for the clarification.